### PR TITLE
Add 58 mainnet

### PR DESCRIPTION
**Pull request by oskarth**
cc @adambabik

### CODE REVIEW

File: ideas/58-mainnet.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    Created: 2017-12-08
+
+
+## Summary
+Alpha builds of Status should default to Mainnet. We should be using Status in a real-world environment.
+
+## Swarm Participants
+
+- Lead Contributor: @adambabik
+- Testing & Evaluation: <!-- @username -->
```
Comment (oskarth) [inline ideas/58-mainnet.md]: We need an evaluator for this one. @annadanchenko?

File: ideas/58-mainnet.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+1. [ ] Security audit is finished by an independent company,
+1. [ ] The final report is available,
+1. [ ] All reported security issues are evaluated and fixed.
+
+## Marketing iteration
+
+Goal Date: 2018-04-??
+
+1. [ ] Status App Bug Bounty program is announced and will last for at least X weeks.
+
```
Comment (oskarth) [inline ideas/58-mainnet.md]: Can we add an exit criteria?

File: ideas/58-mainnet.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+- Lead Contributor: @adambabik
+- Testing & Evaluation: <!-- @username -->
+- Contributor: @themue
+- Contributor: <!-- @username -->
+- PM: @chadyj 
+- UX (if relevant): <!-- @username -->
+
+## Product Overview
+
+Currently the Status alpha builds use testnet, but using Mainnet is a key milestone for the public launch of Status. There will be potentially be major performance and security implications so these issues should be surfaced as soon as possible so the app can be tested in a real-world environment.
```
Comment (themue) [inline ideas/58-mainnet.md]: Always got troubles with weak terms like _"should"_ and _"as soon as possible"_. Better is "have to be surfaced until MVP" with the MVP date mentioned below.

File: ideas/58-mainnet.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+- Contributor: <!-- @username -->
+- PM: @chadyj 
+- UX (if relevant): <!-- @username -->
+
+## Product Overview
+
+Currently the Status alpha builds use testnet, but using Mainnet is a key milestone for the public launch of Status. There will be potentially be major performance and security implications so these issues should be surfaced as soon as possible so the app can be tested in a real-world environment. 
+
+From a users perspective, Mainnet should be used as it is required by most popular DApps, and is needed for real transactions, so the full potential of Status can't be seen until we switch.
+
```
Comment (themue) [inline ideas/58-mainnet.md]: Even if the title "Add Mainnet" says we're adding it I would like to make more clear here that the other networks are still able to select. And it also would be interesting if it is intended that Mainnet will become default or has to be selected explicitly (together with a possible warning) while we're in test.

File: ideas/58-mainnet.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+## Product Description
+
+This is a default setting change, so there aren't any user facing options.
```
Comment (themue) [inline ideas/58-mainnet.md]: Oh, just seen, this answers my question from above. If the user is not facing any option but DApp creators can toggle networks how will it be done? Or does it mean, the user is not facing options by default but any user is able to switch networks?

File: ideas/58-mainnet.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+Some users, such as DApp creators may want to test their app on testnet (or some other use case), so they will be able to toggle networks as can be done currently.
+
+## Requirements & Dependancies
+
+Security and performance should should be discussed and considered.
+
+Also, several Dapps should be tested.
```
Comment (themue) [inline ideas/58-mainnet.md]: We have to make both more concrete. Would appreciate a discussion about it (criteria for security, performance, and reliability as well as a selection of DApps for repeatable tests).

File: ideas/58-mainnet.md (hunk 5 above)
Comment (adambabik) [reply ideas/58-mainnet.md]: I guess it just means that it will be a **default** option. It will still be possible to change the network.

File: ideas/58-mainnet.md (hunk 2 above)
Comment (adambabik) [reply ideas/58-mainnet.md]: I am not sure what exit criteria we have here. Like how long this program will last etc. Do we have Jonny on Github?

File: ideas/58-mainnet.md (hunk 5 above)
Comment (themue) [reply ideas/58-mainnet.md]: That's how I understand it too.  But we could formulate it a bit more clear.

File: ideas/58-mainnet.md (hunk 2 above)
Comment (andytudhope) [reply ideas/58-mainnet.md]: afaik the bug bounty for mainnet was cancelled.

Comment (j-zerah) [reply ideas/58-mainnet.md]: bug bounty for mainnet was cancelled

File: ideas/58-mainnet.md (hunk 3 above)
Comment (adambabik) [reply ideas/58-mainnet.md]: I am not sure how I can rephrase it to make it more clear. I understand it this way that this swarm is about enabling mainnet so that all these issues can be surfaced. I am not sure it's gonna happen exactly after MVP or by the end of the swarm.

Comment (themue) [reply ideas/58-mainnet.md]: Maybe _"One goal is to surface these issues early enough for testing the app in a real-world environment."_?

File: ideas/58-mainnet.md (hunk 1 above)
Comment (oskarth) [reply ideas/58-mainnet.md]: @chadyj can we put you down here?

File: ideas/58-mainnet.md (hunk 3 above)
Comment (adambabik) [reply ideas/58-mainnet.md]: Thanks!

File: ideas/58-mainnet.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+## Minimum Viable Product
+
+**This MVP does not mean that after it's finished we release the app with Mainnet enabled by default but defines prerequisites that should be done before we consider doing so.**
+
+Goal Date: 2018-04-06
+
+1. [ ] Development of required components is done. The work is tracked here: https://github.com/orgs/status-im/projects/15,
+1. [ ] There is a Status App build with mainnet available,
+1. [ ] All components have been tested by the Q&A Team,
+1. [ ] Mainnet is defult in dev/nightly builds.
```
Comment (themue) [inline ideas/58-mainnet.md]: Typo, s/defult/default/.

File: ideas/58-mainnet.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+- Testing & Evaluation: <!-- @username -->
+- Contributor: @themue
+- Contributor: <!-- @username -->
+- PM: @chadyj
+
+## Product Overview
+
+Currently the Status alpha builds use testnet, but using Mainnet is a key milestone for the public launch of Status. There will be potentially major performance and security implications so the goal is to surface these issues early enough by testing the app in a real-world environment.
+
+From a users perspective, Mainnet should be a default network as it is required by most popular DApps, and is needed for real transactions, so the full potential of Status can't be seen until we switch. The other networks will continue to be available for developers and testers.
```
Comment (themue) [inline ideas/58-mainnet.md]: Would rephrase last sentence: _"The other networks will continue to be available, ie for developers and tests."_. It sounds less restrictive.

File: ideas/58-mainnet.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+This is a default setting change, so there aren't any user facing options. The other networks are still visible.
+
+Some users, such as DApp creators may want to test their app on testnet (or some other use case), so they will be able to toggle networks as can be done currently.
+
+## Requirements & Dependancies
+
+Security and performance should should be discussed and considered.
+
+Also, several Dapps should be tested.
```
Comment (themue) [inline ideas/58-mainnet.md]: s/Dapps/DApps/ for consistency.
